ricecooker/classes/nodes.py

```diff
@@ -677,12 +677,12 @@ def __init__(
         role=roles.LEARNER,
         license_description=None,
         copyright_holder=None,
-        grade_levels=None,
-        resource_types=None,
-        learning_activities=None,
-        accessibility_labels=None,
-        categories=None,
-        learner_needs=None,
+        grade_levels=[],
+        resource_types=[],
+        learning_activities=[],
```

Comment:
There's also another gotcha here - you shouldn't use mutable types as default arguments, because they get shared across invocations of the method.

https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

So, these defaults should still be `None` and when you come to assign them you would do:

```
        self.grade_levels = grade_levels or []
        self.resource_types = resource_types or []
        self.learning_activities = learning_activities or []
        self.accessibility_labels = accessibility_labels or []
        self.categories = categories or []
        self.learner_needs = learner_needs or []
```

Reply:
> Python’s default arguments are evaluated once when the function is defined, not each time the function is called (like it is in say, ~Ruby~ _every reasonable programming language_)

:face_with_spiral_eyes: